adapters/tlsio_ssl_esp8266.c

```diff
@@ -0,0 +1,1085 @@
+ * FunctionName : esp_ssl_set_fd
+ * Description  : sent data for client or server
+ * Parameters   : s -- espconn to set for client or server
+ *                fd -- data to send
+ *                length -- length of data to send
+ * Returns      : none
+*******************************************************************************/
+int ICACHE_FLASH_ATTR ERR_get_error(void)
+{
+    return 0;
+}
+
+void ICACHE_FLASH_ATTR ERR_error_string_n(uint32 error, char* out, uint32 olen)
+{
+    return;
+}
+
+void ICACHE_FLASH_ATTR ERR_error_string(uint32 error, char* out)
+{
+    return;
+}
+
+/*this function will clone an option given by name and value*/
+static void* tlsio_openssl_CloneOption(const char* name, const void* value)
+{
+    LogError("rita: tlsio_openssl_CloneOption");
+    void* result;
+    if(
+        (name == NULL) || (value == NULL)
+    )
+    {
+        LogError("invalid parameter detected: const char* name=%p, const void* value=%p", name, value);
+        result = NULL;
+    }
+    else
+    {
+        if (strcmp(name, "TrustedCerts") == 0)
+        {
+            if(mallocAndStrcpy_s((char**)&result, value) != 0)
+            {
+                LogError("unable to mallocAndStrcpy_s TrustedCerts value");
+                result = NULL;
+            }
+            else
+            {
+                /*return as is*/
+            }
+        }
+        else if (strcmp(name, "x509certificate") == 0)
+        {
+            if (mallocAndStrcpy_s((char**)&result, value) != 0)
+            {
+                LogError("unable to mallocAndStrcpy_s x509certificate value");
+                result = NULL;
+            }
+            else
+            {
+                /*return as is*/
+            }
+        }
+        else if (strcmp(name, "x509privatekey") == 0)
+        {
+            if (mallocAndStrcpy_s((char**)&result, value) != 0)
+            {
+                LogError("unable to mallocAndStrcpy_s x509privatekey value");
+                result = NULL;
+            }
+            else
+            {
+                /*return as is*/
+            }
+        }
+        else
+        {
+            LogError("not handled option : %s", name);
+            result = NULL;
+        }
+    }
+    return result;
+}
+
+/*this function destroys an option previously created*/
+static void tlsio_openssl_DestroyOption(const char* name, const void* value)
+{
+    LogError("rita: tlsio_openssl_DestroyOption");
+    /*since all options for this layer are actually string copies., disposing of one is just calling free*/
+    if (
+        (name == NULL) || (value == NULL)
+        )
+    {
+        LogError("invalid parameter detected: const char* name=%p, const void* value=%p", name, value);
+    }
+    else
+    {
+        if (
+            (strcmp(name, "TrustedCerts") == 0) ||
+            (strcmp(name, "x509certificate") == 0) ||
+            (strcmp(name, "x509privatekey") == 0)
+        )
```

Comment:
I'm following the current [implementation of tlsio_openssl.c](https://github.com/Azure/azure-c-shared-utility/blob/develop/src/tlsio_openssl.c#L147). If this needs to be different, can you provide an example? Thanks!